README.md

```diff
@@ -32,22 +33,237 @@ scheduled_task { 'Run Notepad':
+
+~~~ puppet
+scheduled_task { 'Disk Cleanup Daily Repeating':
+  ensure    => 'present',
+  command   => 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe',
+  arguments => '-File "C:\\Scripts\\Clear-DiskSpace.ps1",
+  enabled   => 'true',
+  trigger   => [{
+    'schedule'         => 'daily',
+    'start_time'       => '07:00',
+    'minutes_duration' => '720',   # Specifies the length of time, in minutes, the task is active
+    'minutes_interval' => '60'     # Causes the task to run every hour
+  }],
+  user      => 'system',             # Specifies the account to run the task as
+}
+~~~
+
+The downside to that task is that it causes the cleanup script to run _every_ day, even on the weekends when there is no activity.
+We can instead use a weekly trigger to fix this:
+
+~~~puppet
+scheduled_task { 'Disk Cleanup Weekly Repeating':
+  ensure    => 'present',
+  command   => 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe',
+  arguments => '-File "C:\\Scripts\\Clear-DiskSpace.ps1",
+  enabled   => 'true',
+  trigger   => [{
+    'schedule'         => 'weekly',
+    'start_time'       => '07:00',
+    'day_of_week'      => ['mon', 'tues', 'wed', 'thurs', 'fri'], # Note the absence of Sunday and Monday
+    'minutes_interval' => '60',
+    'minutes_duration' => '720'
+  }],
+  user      => 'system',
+}
+~~~
+
+Similarly, we can schedule our cleanup script to run monthly if we decide the cleanup script doesn't need to run as often or is particularly resource intensive.
+The following example sets the scheduled task to run at 0700 on the first day of the month every month:
+
+~~~puppet
+scheduled_task { 'Disk Cleanup Monthly First Day':
+  ensure    => 'present',
+  command   => 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe',
+  arguments => '-File "C:\\Scripts\\Clear-DiskSpace.ps1",
+  enabled   => 'true',
+  trigger   => [{
+    'schedule'         => 'monthly',
+    'start_time'       => '07:00',
+    'on'               => [1]        # Run every month on the first day of the month.
+  }],
+  user      => 'system',
+}
+~~~
+
+One problem with the monthly trigger above is that we can't guarantee that the cleanup script doesn't run during a work day.
+We can specify the trigger to run on the task on the _first saturday_ of the month instead of the first day:
+
+~~~puppet
+scheduled_task { 'Disk Cleanup Monthly First Saturday':
+  ensure    => 'present',
+  command   => 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe',
+  arguments => '-File "C:\\Scripts\\Clear-DiskSpace.ps1",
+  enabled   => 'true',
+  trigger   => [{
+    'schedule'       => 'monthly',
+    'start_time'     => '07:00',
+    'day_of_week'    => 'sat',     # Specify the day of the week to trigger on
+    'which_occurence => 'first'    # Specify which occurance to trigger on, up to fifth
```

Comment:
Is this only available when using the version 2 api? If so, shouldn't we let users know that this example only works on the modern api?

Reply:
It looks like this is included in the [legacy api](https://puppet.com/docs/puppet/5.4/types/scheduled_task.html#scheduled_task-attribute-trigger) too.

I don't think _at this point_ there's anything net new in the module for the API, it's almost entirely parity / fixes.

Reply:
Interestingly, this is apparently supported in the [original implementation](https://github.com/puppetlabs/puppet/commit/0d7c79727cfb158f7182f791ca3ad6e6248c1c23#diff-1df63dec83110755c93b68353185d035R179), even. 😮